test-app/src/androidTest/java/com/avito/android/ui/test/ButtonsOverRecyclerWithCollapsingToolbarTest.kt

```diff
@@ -0,0 +1,16 @@
+package com.avito.android.ui.test
+
+import com.avito.android.ui.ButtonsOverRecyclerWithCollapsingToolbarActivity
+import org.junit.Rule
+import org.junit.Test
+
+class ButtonsOverRecyclerWithCollapsingToolbarTest {
+
+    @get:Rule
+    val rule = screenRule<ButtonsOverRecyclerWithCollapsingToolbarActivity>(launchActivity = true)
+
+    @Test
+    fun listElement_elementClicked_whenThereIsOverlappedButtonInScreenWithCollapsingToolbar() {
```

Comment:
How do you assure that view is overlapped in first place?

Reply:
Previous scroll logic scrolled to top (or bottom) edge of the screen. This position is always overlapped by buttons that I've created in layout.

Also 90th element is unreachable by software scrolling with expanded app bar.

This test check new scroll logic (to center instead of edge) and collapsing app bars logic.